app/services/pairing_strategies/swiss.rb

```diff
@@ -68,13 +83,13 @@ def auto_score(pairing, player_index)
     end
 
     def players_with_byes
-      return players.with_first_round_bye if first_round?
+      return players.select { |_, v| v.first_round_bye } if first_round?
 
-      []
+      {}
     end
 
     def players_to_pair
-      @players_to_pair ||= players - players_with_byes
+      @players_to_pair ||= players.filter_map { |k, v| v unless players_with_byes.key?(k) }
```

Comment:
The fact that this now returns a map makes a number of the tests very unstable. I think they were already a little unstable, but now the sort order varies much more, which makes it worse. I also notice that everywhere that calls this also calls to_a on the result.

Maybe we should sort the players in an array before returning from here? I'm not sure how else we can control the pairing behaviour in tests.

This is the same in the single sided swiss strategy.

Reply:
in short - we have some existing flaky tests presumably due to ordering issues (and maybe still a few related to test data setup issues) but i haven't seen how this makes it worse yet.

Reply:
The shuffle is controlled in tests with a Random object that's passed in with a fixed seed.

Another way to avoid the randomness would be to pass in the list of players in the test so the order is fixed at that point. That would also speed up the test suite by avoiding the database query in tests that aren't about that.